deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,134 @@
+    leafs of the trained trees.
+    Parameters
+    ----------
+    columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to check, if none are given checks all columns except ignored ones.
+    ignore_columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to ignore, if none given checks based on columns variable
+    n_top_features : int , default: 5
+        Number of features to use for segment search. Top columns are selected based on feature importance.
+    segment_minimum_size_ratio: float , default: 0.05
+        Minimum size ratio for segments. Will only search for segments of
+        size >= segment_minimum_size_ratio * data_size.
+    alternative_scorer : Tuple[str, Union[str, Callable]] , default: None
+        Scorer to use as performance measure, either function or sklearn scorer name.
+        If None, a default scorer (per the model type) will be used.
+    loss_per_sample: Union[np.array, pd.Series, None], default: None
+        Loss per sample used to detect relevant weak segments. If pd.Series the indexes should be similar to those in
+        the dataset object provide, if np.array the order should be based on the index order of the dataset object and
+        if None the check calculates loss per sample by via log loss for classification and MSE for regression.
+    n_samples : int , default: 10_000
+        Maximum number of samples to use for this check.
+    n_to_show : int , default: 3
+        number of segments with the weakest performance to show.
+    categorical_aggregation_threshold : float , default: 0.05
+        In each categorical column, categories with frequency below threshold will be merged into "Other" category.
+    """
+
+    def __init__(self, columns: Union[Hashable, List[Hashable], None] = None,
+                 ignore_columns: Union[Hashable, List[Hashable], None] = None, n_top_features: int = 5,
+                 segment_minimum_size_ratio: float = 0.05, alternative_scorer: Dict[str, Callable] = None,
+                 loss_per_sample: Union[np.ndarray, pd.Series, None] = None, n_samples: int = 10_000,
+                 categorical_aggregation_threshold: float = 0.05, n_to_show: int = 3, **kwargs):
+        super().__init__(**kwargs)
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_top_features = n_top_features
+        self.segment_minimum_size_ratio = segment_minimum_size_ratio
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.loss_per_sample = loss_per_sample
+        self.alternative_scorer = alternative_scorer if alternative_scorer else None
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        text_data = context.get_data_by_kind(dataset_kind)
+        if text_data.meta_data is None:
```

Comment:
additional_data pls :) 
and in the errors too

Reply:
Right!